### ISSUE TITLE
CKV2_GCP_3 seems to be doing the opposite of what it says it is doing

### ISSUE THREAD

**Issue by jbratto** [bridgecrewio/checkov#1416]
**Describe the bug**
Hello, it appears that this check (Ensure there are only GCP-managed service account keys for each service account) is in fact failing unless you actually create a user managed service account key.  Either I'm misunderstanding something, or this check is setup to do the opposite of what it says it is doing.

**To Reproduce**
To demonstrate this, I used the sample code provided by the check documentation and then ran *gcloud iam service-accounts keys list* to show the details about the service account keys.

**Expected behavior**
A google_service_account_key resource in Terraform creates a user managed key, so I would expect this check to be trigger when one is created, rather than when one is not created.

**Desktop (please complete the following information):**
 This example was run from Google Cloud Shell using the latest checkov container (7/25/2021)

**Additional context**

Using this sample Terraform code:
```
provider "google" {}
provider "local" {}

resource "google_service_account" "account_ok" {
  account_id = "pass-account"
}

resource "google_service_account_key" "ok_key" {
  service_account_id = google_service_account.account_ok.name
}

resource "local_file" "foo" {
    content     = google_service_account_key.ok_key.private_key
    filename    = "${path.module}/key.json"
}
```

Executing checkov gets the following results:
```
docker run --tty --volume /home/xxxxxxx/test:/tf bridgecrew/checkov --directory /tf
terraform scan results:

Passed checks: 1, Failed checks: 0, Skipped checks: 0
```

However, if I check for details on the service account key, I see that there are now actually 2 service account keys. One is managed by the system, and the second one is a user managed key. 
```
xxxxxx@cloudshell:~/test (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=pass-account@xxxxxx.iam.gserviceaccount.com
KEY_ID                                    CREATED_AT            EXPIRES_AT
e6af1e75f8a8801b34440951edd76365414e2147  2021-07-25T14:59:49Z  9999-12-31T23:59:59Z
8c635682666032bd3c215fd47f41db0f1bda745e  2021-07-25T14:59:49Z  2023-07-26T05:13:00Z
xxxxxx@cloudshell:~/test (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=pass-account@xxxxxx.iam.gserviceaccount.com --managed-by=user
KEY_ID                                    CREATED_AT            EXPIRES_AT
e6af1e75f8a8801b34440951edd76365414e2147  2021-07-25T14:59:49Z  9999-12-31T23:59:59Z
xxxxxx@cloudshell:~/test2 (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=pass-account@xxxxxx.iam.gserviceaccount.com --managed-by=system
KEY_ID                                    CREATED_AT            EXPIRES_AT
8c635682666032bd3c215fd47f41db0f1bda745e  2021-07-25T14:59:49Z  2023-07-26T05:13:00Z
```

For a second example, I just create a service account without a key.
```
provider "google" {}

resource "google_service_account" "account_bad" {
  account_id = "fail-account"
}
```

Checkov flags this because of CKV2_GCP_3.

```
docker run --tty --volume /home/xxxxxxxx/test2:/tf bridgecrew/checkov --directory /tf
terraform scan results:

Passed checks: 0, Failed checks: 1, Skipped checks: 0

Check: CKV2_GCP_3: "Ensure that there are only GCP-managed service account keys for each service account"
        FAILED for resource: google_service_account.account_bad
```

However, if I check for service account keys, there is only a system managed key for this service account

```
xxxxxx@cloudshell:~/test2 (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=fail-account@xxxxxx.iam.gserviceaccount.com
KEY_ID                                    CREATED_AT            EXPIRES_AT
bba135e65c69aec6ed974d6db7c1607f211d9d05  2021-07-25T15:15:19Z  2023-08-15T21:24:33Z
xxxxxx@cloudshell:~/test2 (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=fail-account@xxxxxx.iam.gserviceaccount.com --managed-by user
Listed 0 items.
xxxxxx@cloudshell:~/test2 (xxxxxx)$ gcloud iam service-accounts keys list --iam-account=fail-account@xxxxxx.iam.gserviceaccount.com --managed-by system
KEY_ID                                    CREATED_AT            EXPIRES_AT
bba135e65c69aec6ed974d6db7c1607f211d9d05  2021-07-25T15:15:19Z  2023-08-15T21:24:33Z
```

**Comment by mzwennes** [comment]
@spothound @jbratto See https://github.com/bridgecrewio/checkov/pull/1884.

**Comment by calexandre** [comment]
I'm also being questioned by our security team regarding this check.
It does not make any sense for checkov to recommend the creation of the service-account-keys to pass the security check.

I believe this is an **urgent** issue, because checkov might be recommending the application of a bad-practice.

**Comment by gruebel** [comment]
thanks to @mzwennes it is fixed in version `2.0.563` 💪
